src/includes/class_reservationPermissions.php

```diff
@@ -1,76 +1,62 @@
 <?php
 class reservationPermissions {
 
-  public function getRecords($id = null){
-      try {
-          // call engine
-          $engine    = EngineAPI::singleton();
-          $localvars = localvars::getInstance();
-          $db        = db::get($localvars->get('dbConnectionName'));
-          $sql       = "SELECT * FROM `reservePermissions`";
-          $validate  = new validate;
+  private $localvars;
+  private $engine;
+  private $this->db;
+  private $validate;
+
+  function __construct() {
+    $this->engine    = EngineAPI::singleton();
+    $this->localvars = localvars::getInstance();
+    $this->db        = db::get($this->localvars->get('dbConnectionName'));
+    $this->validate  = new validate;
+  }
 
-          if (isset($id)) {
-           $id        = dbSanitize($id);
-          }
+  public function getRecords($id = null){
+    $sql = "SELECT * FROM `reservePermissions`";
+    try {
+      if(!isnull($id) && !$this->validate->integer($id)){
+        throw new Exception("Invalid ID provided.");
+      }
 
-          // test to see if Id is present and valid
-          if(!isnull($id) && $validate->integer($id)){
-              $sql .= sprintf('WHERE id = %s LIMIT 1', $id);
-          }
+      if(!isnull($id)){
+        $sql .= "WHERE id = ? LIMIT 1";
+      }
 
-          // if no valid id throw an exception
-          if(!$validate->integer($id) && !isnull($id)){
-              throw new Exception("I don't want to be tried!");
-          }
+      $sqlResult = $this->db->query($sql,array($id));
 
-          // get the results of the query
-          $sqlResult = $db->query($sql);
-          // if return no results
-          // else return the data
-          if ($sqlResult->error()) {
-              throw new Exception("ERROR SQL" . $sqlResult->errorMsg());
-          }
-          if ($sqlResult->rowCount() < 1) {
-             return "There are no records in the database.";
-          }
-          else {
-              $data = array();
-              while($row = $sqlResult->fetch()){
-                  $data[] = $row;
-              }
-              return $data;
-          }
-      } catch (Exception $e) {
-          errorHandle::errorMsg($e->getMessage());
+      if ($sqlResult->error()) {
+        throw new Exception("ERROR SQL" . $sqlResult->errorMsg());
       }
+
+      return $sqlResult->fetchAll();
+    } catch (Exception $e) {
+      errorHandle::newError(__METHOD__."() - ".$e->getMessage, errorHandle::DEBUG);
+      return false;
+    }
```

Comment:
On error, you need to return false, otherwise, you have no idea that something went wrong and processing should stop elsewhere.